code/game/machinery/big_manipulator.dm

```diff
@@ -0,0 +1,264 @@
+			power_use_lvl = 0.2
+			set_greyscale(COLOR_YELLOW)
+			manipulator_hand?.set_greyscale(COLOR_YELLOW)
+		if(2)
+			working_speed = 1.4 SECONDS
+			power_use_lvl = 0.4
+			set_greyscale(COLOR_ORANGE)
+			manipulator_hand?.set_greyscale(COLOR_ORANGE)
+		if(3)
+			working_speed = 0.8 SECONDS
+			power_use_lvl = 0.6
+			set_greyscale(COLOR_RED)
+			manipulator_hand?.set_greyscale(COLOR_RED)
+		if(4)
+			working_speed = 0.2 SECONDS
+			power_use_lvl = 0.8
+			set_greyscale(COLOR_PURPLE)
+			manipulator_hand?.set_greyscale(COLOR_PURPLE)
+
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * power_use_lvl
+
+/// Changing take and drop turf tiles when we anchore manipulator or if manipulator not in turf.
+/obj/machinery/big_manipulator/proc/take_and_drop_turfs_check()
+	if(anchored && isturf(src.loc))
+		take_turf = get_step(src, take_here)
+		drop_turf = get_step(src, drop_here)
+	else
+		take_turf = null
+		drop_turf = null
+
+/// Changing take and drop turf dirs and also changing manipulator hand sprite dir.
+/obj/machinery/big_manipulator/proc/rotate_big_hand()
+	switch(take_here)
+		if(NORTH)
+			take_here = EAST
+			drop_here = WEST
+		if(EAST)
+			take_here = SOUTH
+			drop_here = NORTH
+		if(SOUTH)
+			take_here = WEST
+			drop_here = EAST
+		if(WEST)
+			take_here = NORTH
+			drop_here = SOUTH
+	manipulator_hand.dir = take_here
+	take_and_drop_turfs_check()
+
+/// Deliting hand will destroy our manipulator core.
+/obj/machinery/big_manipulator/proc/on_hand_qdel()
+	SIGNAL_HANDLER
+
+	deconstruct(TRUE)
+
+/// Pre take and drop proc from [take and drop procs loop]:
+/// Check if we have item on take_turf to start take and drop loop
+/obj/machinery/big_manipulator/proc/is_work_check()
+	for(var/obj/item/take_item in take_turf.contents)
+		try_take_thing(take_turf, take_item)
+		break
+
+/// First take and drop proc from [take and drop procs loop]:
+/// Check if we can take item from take_turf to work with him. This proc also calling from ATOM_ENTERED signal.
+/obj/machinery/big_manipulator/proc/try_take_thing(datum/source, atom/movable/target)
+	SIGNAL_HANDLER
+
+	if(!on)
+		return
+	if(!anchored)
+		return
+	if(QDELETED(source) || QDELETED(target))
+		return
+	if(on_work)
+		return
+	if(!use_energy(active_power_usage, force = FALSE))
+		on = FALSE
+		say("Not enough energy!")
+		return
+	if(isitem(target))
+		start_work(target)
+
+/// Second take and drop proc from [take and drop procs loop]:
+/// Taking our item and start manipulator hand rotate animation.
+/obj/machinery/big_manipulator/proc/start_work(atom/movable/target)
+	target.forceMove(src)
+	containment_obj = WEAKREF(target)
+	on_work = TRUE
+	do_rotate_animation(1)
+	addtimer(CALLBACK(src, PROC_REF(drop_thing), target), working_speed)
+
+/// Third take and drop proc from [take and drop procs loop]:
+/// Drop our item and start manipulator hand backward animation.
+/obj/machinery/big_manipulator/proc/drop_thing(atom/movable/target)
+	target.forceMove(drop_turf)
```

Comment:
i'd add some form of repeating check here in case the area can't accept the item being dropped off ie bouldertech things

Reply:
Don't understand

Reply:
basically add a proc that checks if the location can accept the item, as if you drop an item onto a full boulder tech it just stays ontop forever iirc so adding a check that goes: can I drop off? if not drop item back off

Reply:
example of what I mean can be found in the file on my pr here: https://github.com/Monkestation/Monkestation2.0/blob/940d18896229d95882024333cf606facc13eee0e/monkestation/code/modules/factory_type_beat/machinery/grabber.dm

Reply:
grabber 🤔

Reply:
Now I understand what you meant. As I see, you have already upgraded the manipulators on the monkey station so that they accept not only items. Since my manipulators don’t know how to do this, I don’t really need this check. However, I added a check to make sure he doesn't put things through walls. If you don't mind, I can (steal) borrow your code to improve the manipulators, or you can do it yourself.